### PR TITLE
Showing the pending pipelines on top

### LINKED ISSUE #1358: Pipeline feed: show pending pipelines (running, needs approval, waiting) on top

### PR DESCRIPTION
**Pull request by bitethecode**
Trying to close #1358

@qwerty287 This should be now clean..!

### CODE REVIEW

**Comment by lafriks** [comment]
Imho this should be switchable (like auto scroll or not for log output) as for me personally this would make it hard to understand step order especially when groups are used

**Comment by bitethecode** [comment]
Could you elaborate what it means `switchable`? I thought the method should behave similar to what we cancel?

**Comment by bitethecode** [comment]
@lafriks Oh, got ya. Any design in mind? or flexible?

**Comment by lafriks** [comment]
Probably on top of the pipeline list there could be icon to switch that

**Comment by bitethecode** [comment]
@lafriks thanks for thought!

**Comment by 6543** [comment]
tell if it's ready to review :rocket:

**Comment by bitethecode** [comment]
@6543 Thanks for checking in. I wasn't able to run the project locally for a while and figured it out this morning. I will update it this week..!

**Comment by 6543** [comment]
if you dont wana setup a local dev environment, we have gitpod integration ...

**Comment by 6543** [comment]
(I personally still prefer a local setup - but just if you did not noticed)

File: web/src/utils/helpers.ts [hunk 1]
```diff
@@ -47,6 +47,19 @@ export function comparePipelines(a: Pipeline, b: Pipeline): number {
...
 
+/**
+ * Compare two pipelines by the status.
+ * Giving pending, running, or started higher priority than other status
+ * @param {Object} a - A pipeline.
+ * @param {Object} b - A pipeline.
+ * @returns {number}
+ */
+export function comparePipelinesWithStatus(a: Pipeline, b: Pipeline): number {
+  const bPrioirty = ['pending', 'running', 'started'].includes(b.status) ? 1 : 0;
```
[the remark this answers is not in the document]
Comment (bitethecode) [reply web/src/utils/helpers.ts:58]: @mscherer nice catch. I will update

**Comment by bitethecode** [comment]
@lafriks @6543 I do have a question if we want to have a button for reloading the sorted pipelines. Currently, we are using `provide` and it's not for updating state like props. So I want to clarify whether we want to take button approaches or not in the first place before refactoring.

**Review by xoxys — APPROVED** [review]

File: web/src/utils/helpers.ts [hunk 2]
```diff
@@ -53,6 +53,19 @@ export function comparePipelines(a: Pipeline, b: Pipeline): number {
...
 }
 
+/**
+ * Compare two pipelines by the status.
+ * Giving pending, running, or started higher priority than other status
+ * @param {Object} a - A pipeline.
+ * @param {Object} b - A pipeline.
+ * @returns {number}
+ */
+export function comparePipelinesWithStatus(a: Pipeline, b: Pipeline): number {
```
Comment (6543) [inline web/src/utils/helpers.ts:63]: do we have a testframework ?

That should have a simple coverage :)

Comment (xoxys) [reply web/src/utils/helpers.ts:63]: Can we merge it or do we want to add one first?

Comment (6543) [reply web/src/utils/helpers.ts:63]: -> #2694 

Input is welcome :)

**Pull request merged**